### PR TITLE
PLANET-6892 Add localize action slug option

### PR DESCRIPTION
**Pull request by sagarsdeshmukh**
Allow admin to localize action slug from permalinks setting page

Ref. https://jira.greenpeace.org/browse/PLANET-6892

**eg.**
[Permalinks setting page](https://www-dev.greenpeace.org/test-phoebe/wp-admin/options-permalink.php)
[Action page](https://www-dev.greenpeace.org/test-phoebe/custom-slug/lorem-ipsum/)

**Testing:**
- Go to actions [listing page](https://www-dev.greenpeace.org/test-phoebe/wp-admin/edit.php?post_type=p4_action)
- Add new action page, view it on frontend, also please check the URL slug
- Now change the custom action slug from [permalinks settings page](https://www-dev.greenpeace.org/test-phoebe/wp-admin/options-permalink.php)
- Go to [actions listing page ](https://www-dev.greenpeace.org/test-phoebe/wp-admin/edit.php?post_type=p4_action) , check different action by clicking on "view" anchor link
- The URL should have the custom action slug in it

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [phoebe](https://www-dev.greenpeace.org/test-phoebe) | [admin](https://www-dev.greenpeace.org/test-phoebe/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-phoebe/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-phoebe) | [composer-local.json](https://github.com/greenpeace/planet4-test-phoebe/blob/main/composer-local.json)

:watch: 2022.11.11 07:56:43

File: src/ActionPage.php [hunk 1]
```diff
@@ -159,4 +165,27 @@ public function register_post_meta() {
...
+		add_settings_field( 'p4_action_posttype_slug', __( 'Action post type slug', 'planet4-master-theme-backend' ), [ $this, 'add_action_slug_field' ], 'permalink', 'optional' );
+	}
+
+	/**
+	 * Add Action slug text field on permalinks page.
+	 */
+	public function add_action_slug_field() {
+
+		$value = get_option( 'p4_action_posttype_slug' );
+		echo '<input type="text" value="' . esc_attr( $value ) . '" name="p4_action_posttype_slug" id="p4_action_posttype_slug" class="regular-text" /><p>' . esc_html__( 'The default Action post type slug is "action".', 'planet4-master-theme-backend' ) . '</p>';
```
Comment (GP-Dan-Tovbein) [inline src/ActionPage.php:189]: Just curious, why do you use `_` instead of `-` to the `name` and `id` fields?

Comment (sagarsdeshmukh) [reply src/ActionPage.php:189]: Nothing specific, I use the same name (`p4_action_posttype_slug`) for wp option setting key hence  got added for name & id, but if we need we can change it.

Comment (GP-Dan-Tovbein) [reply src/ActionPage.php:189]: It's just to follow the same rule of our templates, but not required to change it

**Review by GP-Dan-Tovbein — APPROVED** [review]
Code looks good to me! I've tested it and working fine! Good job 👏

**Pull request merged**